src/realm/array_string.hpp

```diff
@@ -166,7 +166,7 @@ class QueryState<StringData> : public QueryStateBase {
     template <Action action>
     bool uses_val()
     {
-        return (action == act_Count);
+        return false;
```

Comment:
I can't see why we were using the value for count anywhere for any data type so this is a nice optimization everywhere 💯